src/SwedbankPay.Sdk.Infrastructure/JsonSerialization/Converters/CustomHttpResponseExceptionConverter.cs

```diff
@@ -0,0 +1,89 @@
+﻿using SwedbankPay.Sdk.Exceptions;
+using System;
+using System.Net.Http;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SwedbankPay.Sdk.JsonSerialization.Converters
+{
+    internal class CustomHttpResponseExceptionConverter : JsonConverter<HttpResponseException>
```

Comment:
It is used in reading `HttpResponseException` before it is converted to `Problem`. Without this we get the following exception:
`System.Text.Json.JsonException : A possible object cycle was detected which is not supported. This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 64.`

Reply:
But why are we trying to serialize or deserialize `HttpResponseException` at all?

Reply:
Not sure, there is a unit test for this, but it might be a leftover from before the re-write.
Shall I delete the test?

Reply:
Which unit test is that?

Reply:
`ProblemResponseException_CanBeSerialized_AsJson` in `src/SwedbankPay.Sdk.Tests/UnitTests/HttpClientExtensionsTests.cs`

Reply:
So the only part of the code that assumes `HttpResponseException` can be serialized is a test? If so, delete it.

Reply:
Done